src/QirRuntime/lib/QIR/context.cpp

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <assert.h>
+
+#include "context.hpp"
+
+#include "CoreTypes.hpp"
+#include "QuantumApi_I.hpp"
+#include "allocationsTracker.hpp"
+
+#ifdef _WIN32
+#define QIR_SHARED_API __declspec(dllexport)
+#else
+#define QIR_SHARED_API
+#endif
+
+extern "C" QIR_SHARED_API Result ResultOne = nullptr;
+extern "C" QIR_SHARED_API Result ResultZero = nullptr;

```

Comment:
Minor:
I don't have experience in writing DLLs, but these two variables seem to me a part of the API exported/exposed by the DLL. In which case it probably makes sense to mention them in the corresponding .hpp file, and providing comments about what they are.

Reply:
The constants are defined in generated QIR and are troublesome. I remember bringing it up in the past but cannot find any bugs on this...

Reply:
Do you have any suggestions for how to best expose the results constants that the QIR relies on?

Reply:
My preference would be to wrap those in function calls, something like: `%Result* __quantum__rt__result_zero()`.